src/main/java/wolox/training/models/User.java

```diff
@@ -74,4 +78,26 @@ public void removeBook(Book book){
         books.remove(book);
     }
 
+    public void setUserName(String userName) {
+        Preconditions.checkNotNull(userName,String.format(Constants.MESSAGE_CHECK_IS_NULL,"userName"));
+        Preconditions.checkArgument(!userName.isEmpty(),String.format(Constants.MESSAGE_CHECK_IS_EMPTY,"userName"));
+        this.userName = userName;
+    }
+
+    public void setName(String name) {
+        Preconditions.checkNotNull(name,String.format(Constants.MESSAGE_CHECK_IS_NULL,"name"));
+        Preconditions.checkArgument(!name.isEmpty(),String.format(Constants.MESSAGE_CHECK_IS_EMPTY,"name"));
+        this.name = name;
+    }
+
+    public void setBirthdate(LocalDate birthdate) {
+        Preconditions.checkNotNull(birthdate,String.format(Constants.MESSAGE_CHECK_IS_NULL,"birthdate"));
```

Comment:
Here I forgot that you also need to validate that it is not empty because Jackson internally call `LocalDate.parse(string)` and that will throws a `java.time.format.DateTimeParseException`